modelforge/dataset/dataset.py

```diff
@@ -16,6 +16,47 @@
     from modelforge.potential.processing import AtomicSelfEnergies
 
 
+# __all__ = [
+#     "BatchData",
+#     "HDF5Dataset",
+#     "NNPInput",
+#     "Metadata",
+#     "TorchDataset",
+#     "DatasetParameters",
+# ]
+
+from pydantic import BaseModel, field_validator, ConfigDict
+
+
+class DatasetParameters(BaseModel):
+    """
+    Class to hold the dataset parameters.
+
+    Args:
+        dataset_name (str): The name of the dataset.
+        version_select (str): The version of the dataset to use
+        num_workers (int): The number of workers to use for the DataLoader
+        pin_memory (bool): Whether to pin memory for the DataLoader
+
+    """
+
+    model_config = ConfigDict(
+        use_enum_values=True, arbitrary_types_allowed=True, validate_assignment=True
+    )
+
+    dataset_name: str
+    version_select: str
+    num_workers: int
+    pin_memory: bool
+
+    @field_validator("num_workers")
+    @classmethod
+    def check_num_workers(cls, v):
+        if v < 1:
+            raise ValueError("num_workers must be greater than or equal to 1")
+        return v
+
```

Comment:
Good suggestion.  In Pydantic we can just use `num_workers: int= Field(gt=0)`. I had this as a user defined one because initially I had something in here to ensure that num_workers was less than the number of cores on the machine, but then realized that isn't actually necessary (it doesn't throw an error if you give it an absurd number of threads, like can sometimes happen with certain mpi implementations).